README.md

```diff
@@ -61,6 +61,64 @@ Or with Nix 2.4+:
     $ nix profile install github:cachix/cachix/latest
 ```
 
+### Dev setup for VS Code
+
+1. [Enable flakes](https://nixos.wiki/wiki/Flakes#Enable_flakes)
+
+2. Set up [direnv](https://nix.dev/tutorials/declarative-and-reproducible-developer-environments#setting-up-direnv) - complete steps `1` and `2`
+
+3. Clone and step into this repo
+  ```sh
+  git clone https://github.com/cachix/cachix
+  cd cachix
+  ```
+
+4. Try using cache
+  ```sh
+  cachix use cachix-shell
+  ```
+
+5. Otherwise, build yourself (for the first time)
+  ```sh
+  direnv allow
+  ```
+
+6. Install VS Code extensions
+  ```sh
+  code --install-extension mkhl.direnv --install-extension haskell.haskell
+  code .
+  ```
+
+7. Hover over a term in e.g. [cachix/src/Cachix/Client/WatchStore.hs](cachix/src/Cachix/Client/WatchStore.hs) to check if HLS works. It may take some time
+
+8. In case of any problems try to reload window (`Ctrl` + `Shift` + `P` -> `Developer: Reload window`)
+
+### Shell dev tools
+
+* To use just the tools from `flake.nix`, run
+  ```sh
+  nix develop
+  ```
+
+* To also use the tools from `shell.nix`, run
+  ```sh
+  nix-shell
```

Comment:
`nix-shell` is the only one that should be suggested here - as it's supported by all Nix versions and does the right thing.

Reply:
But the point is to also use flakes (see 1.)